lib/riemann/helpers/event.ex

```diff
@@ -59,7 +40,46 @@ defmodule Riemann.Helpers.Event do
         |> Map.delete(:metric_sint64)
         |> Map.put(:attributes, attributes)
       end
+    end
+  end
+
+  @moduledoc false
+  def build(args, mod) do
+    args
+    |> Enum.into(%{})
+    |> Map.put_new(:time, :erlang.system_time(:seconds))
+    |> Map.put_new_lazy(:host, &default_event_host/0)
+    |> set_attributes_field
+    |> set_metric_pb_fields
+    |> Map.to_list
+    |> mod.new()
+  end
 
+  defp default_event_host do
+    event_host_setting() || machine_hostname()
+  end
+
+  defp set_attributes_field(map) do
```

Comment:
The tricky part to pattern matching this is the `:attributes` field can either be `nil` or not set. If it's nil or not set, we return the map, if it's anything else send it to `Attribute.build/1`. If I wanted to use pattern matching without any other adjustments to the code, it might have to look something like this to keep the same behavior:
```elixir
defp set_attributes_field(%{attributes: nil}=map), do: map
defp set_attributes_field(%{attributes: a}=map) do
  Map.put(map, :attributes, Attribute.build(a))
end
defp set_attributes_field(map), do: map
```

 To make this prettier, I'm thinking I could update `build/2` to default that attribute to `nil`:
```elixir
def build(args, mod) do
  args
  |> Enum.into(%{})
  |> Map.put_new(:attributes, nil)
  |> # ...
end
```
Then I could pattern match like this:
```elixir
defp set_attributes_field(%{attributes: nil}=map), do: map
defp set_attributes_field(%{attributes: a}=map) do
  Map.put(map, :attributes, Attribute.build(a))
end
```

I took the second approach with the `set_metric_pb_fields` function on my next PR. What do you think?